src/utils/utils.jl

```diff
@@ -78,3 +121,25 @@ function check_kwargs_psid(; kwargs...)
     psid_kwargs = filter(x -> in(first(x), ACCEPTED_PSID_TEST_SYSTEMS_KWARGS), kwargs)
     return psid_kwargs
 end
+
+"""
+Creates a JSON file informing the user about the meaning of the hash value in the file path
+if it doesn't exist already 
+"""
+function check_parameters_json(case_args::Dict{Symbol, <:Any})
+    dir_path = get_serialization_dir(case_args)
+    file_path = joinpath(dir_path, "case_parameters.json")
+    case_args_json = JSON.json(case_args)
+
+    if !isfile(file_path)
+        open(file_path, "w") do file
+            write(file, case_args_json)
+        end
+    end
+end
+
+function has_duplicates(directory::String, filename::String)
+    files = readdir(directory)
+    file_count = count(file -> file == filename, files)
+    return file_count > 1
```

Comment:
How can there be more than one file with the same name? Don’t you just want to check if the file exists?
```
isfile(joinpath(directory, filename))
```

Reply:
You suggested we add this function because we use file names in the file path but never enforces file names to be unique. This function is harmless. I would keep it as a security measure.

Reply:
I suggested adding a check to prevent an overwrite of a duplicate filename. That is not what this code does. It looks for the presence of two or more files with the same name in the directory, which is impossible. I think you want the line of code I typed above.